pandas/tests/io/parser/quoting.py

```diff
@@ -151,3 +152,19 @@ def test_quotechar_unicode(self):
         if PY3:
             result = self.read_csv(StringIO(data), quotechar=u('\u0001'))
             tm.assert_frame_equal(result, expected)
+
+    def test_unbalanced_quoting(self):
+        # see gh-22789.
+        data = "a,b,c\n1,2,\"3"
+
+        if self.engine == "c":
+            regex = "EOF inside string starting at row 1"
```

Comment:
were these the same before?

Reply:
no what i mean is why are the messages different on c / python engines?

Reply:
Because we have no control over what Python's CSV library does and the errors that it raises when parsing a CSV.  This difference was actually there before, which is why I ultimately why I made the removal that we discussed [above](https://github.com/pandas-dev/pandas/pull/22814#discussion_r220181152).

Reply:
ok great